### ISSUE TITLE
Increase landing page loading speed

### ISSUE THREAD

**Issue by jumbosushi** [ubyssey/ubyssey-dispatch-theme#562]
Currently, our landing page loads pretty slow.

![image](https://user-images.githubusercontent.com/9669739/46170055-9869a500-c251-11e8-8b35-f73af587ab53.png)

Use [Google Speed Insights](https://developers.google.com/speed/pagespeed/insights/) to check how you're making progress. 

Couple ideas to make this happen:
- django query isn't efficient ([django-debug-toolbar](https://github.com/jazzband/django-debug-toolbar) can be used to analyze the queries)
- lazy load the front end images
- better Image compression
- cache "most popular this week" for few hours

**Comment by b1s0ndev** [comment]
I can start working on this

**Comment by b1s0ndev** [comment]
After looking into some areas for optimization I came across a little bit of a blocker:

1. Each of the sections seem to be using the medium sized images (even though they are thumbnail size) as you can see [here](https://drive.google.com/open?id=1A9CHQ_H8nmEboO4R7iI-0A0YpK23wvEn)

2. So I tried looking in to see if I can change these images to use the thumbnail ones instead but it seems like the page is already trying to achieve that:

- This is what renders the article section:
```
<div class="u-flex--tablet">
      {% for article in articles.rest %}
        {% include 'objects/articles/column.html' with article=article padded=True %}
      {% endfor %}
  </div>
```

- Inside column.html we have:
`<a class="o-article__image" href="{{ article.get_absolute_url }}" style="background-image: url('{{ article.featured_image.image.get_thumbnail_url }}');"></a>`

So as you can see, "get_thumbnail_url" still seems to be returning a "medium" image even though it tries to retrieve a thumbnail url. So I looked into the dispatch codebase and tried to see if there was anything wrong with "get_thumbnal_url":

```
def get_thumbnail_url(self):
        """Returns the thumbnail URL."""
        return '%s%s-%s.jpg' % (settings.MEDIA_URL, self.get_name(), self.THUMBNAIL_SIZE)
```

So this function seems to be producing the correct behaviour (returning a thumbnail image), but we still seem to get medium sized images on the front end.

I'm not too sure how to debug this locally because I don't have the data to fill in the appropriate sections, is there some data I can load locally to look into this further (Ie: data for any section such as "Features" or "Sports")? Or do you happen to know why it might be producing this behaviour? Am I looking at the wrong file?

TL;DR - The front end is loading medium sized images even tho we are calling the "get_thumbnail_url" function. Ultimately if we can load thumbnail images for each section, we have potential to reduce the image size by at least 50%

@jumbosushi @Rowansdabomb

**Comment by Rowansdabomb** [comment]
If we are grabbing the thumbnail url, but still getting a medium sized image, then it maybe that the thumbnail image size is never created when the image is originally saved.

**Comment by b1s0ndev** [comment]
@Rowansdabomb Is an image considered "saved" if we are able to locate the URL in the storage api?
Here is an example of an image that should be using thumbnail but is a [medium image](https://ubyssey.storage.googleapis.com/media/images/2018/10/cascadia_project_poster-medium.jpg)

However I was still able to find the [thumbnail image](https://ubyssey.storage.googleapis.com/media/images/2018/10/cascadia_project_poster-square.jpg) by changing the url suffix.

But the medium image is still being displayed on the front end

**Comment by Rowansdabomb** [comment]
If you can locate the image URL, that should mean it was uploaded and saved. The save function for an image can be found [here](https://github.com/ubyssey/dispatch/blob/db8902e069e5a2c631fd2ed4842d9cd8ec796c3c/dispatch/modules/content/models.py#L517).
